### PR TITLE
Replace travis-ci with GitHub actions

### PR DESCRIPTION
**Pull request by julianoes**
I'm not sure if the deploy stuff is still used and have ignored it.

### CODE REVIEW

**Review by TSC21 — APPROVED** [review]

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: Install dependencies
+        run: sudo apt-get update && sudo apt-get install -y build-essential wget astyle
+      - name: Get toolchain
+        run: |
+            wget https://armkeil.blob.core.windows.net/developer/Files/downloads/gnu-rm/9-2020q2/gcc-arm-none-eabi-9-2020-q2-update-x86_64-linux.tar.bz2
+            tar -jxf gcc-arm-none-eabi-7-2018-q2-update-linux.tar.bz2
```
Comment (davids5) [inline .github/workflows/main.yaml:25]: wget file not the un-tar-ed one

**Review by davids5 — CHANGES_REQUESTED** [review]

**Comment by dagar** [comment]
Build failing.

Once this is working it can be updated to store the artifacts and then the Jenkinsfile can be deleted.

**Comment by julianoes** [comment]
> Build failing.

I can see that :smile:.

```
Makefile:69: Your toolchain doesn't support -mcpu=cortex-m7.
Please use gcc-arm-embedded 4.8 2014q3 or newer. Skipping this sub-library.
```

That's with arm-none-eabi-gcc 9.3.1. We would need this one: https://github.com/libopencm3/libopencm3/commit/cccc1267c5d14105aec94e0606d184a2f2dd2008

**Review by davids5 — CHANGES_REQUESTED** [review]
Depends on https://github.com/PX4/PX4-Bootloader/pull/197 coming in first. So the code for the IO fits with GCC 9

**Comment by julianoes** [comment]
I've tested this together with #197:

> px4fmu-v2: flashing 2x worked, -b works, flashing IO worked 3x,
px4fmu-v4: flashing 2x worked, -b works
px4fmu-v5: flashing FMU 2x worked, -b works, flashing IO 3x

**Review by davids5 — APPROVED** [review]

**Pull request merged**